### ISSUE TITLE
Unexpected output attempting to run vlan-switch test

### ISSUE THREAD

**Issue by jafingerhut** [oxidecomputer/p4#58]
Systems I tried this on, and got the same results:
+ Ubuntu 24.04, aarch64, VM running in VirtualBox on an M3 Apple Silicon Mac
+ Ubuntu 20.04 x86_64 VM rented from AWS

I was following instructions here https://oxidecomputer.github.io/p4/book/02-01-vlan-switch.html to try running the test for the vlan-switch program, but it hung waiting for the first packet to come out:

```
$ p4/book/code/src/bin
$ cargo run --bin vlan-switch
   Compiling x4c-book v0.1.0 (/home/andy/p4/book/code)
    Finished dev [unoptimized + debuginfo] target(s) in 0.96s
     Running `/home/andy/p4/target/debug/vlan-switch`
[
    TableEntry {
        action_id: "filter",
        keyset_data: [
            0,
            0,
        ],
        parameter_data: [
            0,
            47,
        ],
    },
]
dbg vlan-switch.rs just before expect_frames
dbg do_expect_frames got here #1 before phy.recv
^C
```

The two 'dbg' lines were from a couple of `println!` calls I added to see where things were hanging.  It is hanging during the call to `expect_frames` in vlan-switch.rs, and inside of the implementation of `do_expect_frames` it is hanging during the call to `phy.recv`.  Thus it appears that no packet is being transmitted by the P4 program.  Perhaps it is being dropped?

**Comment by rcgoodfellow** [comment]
The book code had bit-rotted a bit. I've addressed this in 

- #61

**Comment by jafingerhut** [comment]
Fixed with PR #61.  Closing.
